main.py

```diff
@@ -48,11 +49,20 @@ def _real_main(config: Config) -> None:
     uploader = Uploader(config)
     dockerizer = Dockerizer(config)
 
+    do_good_files = good_files(config)
+
     gitter.start_gitting()
+
     builder.start_building()
     uploader.start_upload()
     dockerizer.start_dockering()
 
+    if (config.do_good_files):
```

Comment:
this complexity doesn't belong in the main file. From the main we just call the start method of each action.

Reply:
Part of the issue that needs to jump between upload and good files

Reply:
and gitter